src/Build/BackEnd/BuildManager/ProjectIsolationMode.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Build.Execution
+{
+    /// <summary>
+    /// The isolation mode to use.
+    /// </summary>
+    public enum ProjectIsolationMode
+    {
+        /// <summary>
+        /// Do not enable isolation.
+        /// </summary>
+        False,
+
+        /// <summary>
+        /// Enable isolation and log isolation violations as messages.
+        /// </summary>
+        /// <remarks>
+        /// Under this mode, only the results from specific (usually
```

Comment:
Would it make sense to stipulate in `MessageUponIsolationViolation` mode that _no_ targets are serialized if no targets are provided in the `-t` switch? I was thinking that, if no targets are provided in `-t`, then maybe just serialize the default targets. The added benefit with the `-t` stipulation is that it adds more clarity to what `TargetResult`s will be serialized.

Reply:
I think "no targets" == "default targets" is a must, since IIRC the normal `ProjectReference` behavior actually is to use the default targets and not "Build" explicitly.